app/routes.py

```diff
@@ -1,2 +1,226 @@
+    data_types = {"title":str, "description":str}
+    completed_at = input.get("completed_at")
+    for name, val_type in data_types.items():
+        if type(input.get(name)) != val_type:
+            return False
+    if ("completed_at" not in input.keys()):
+        return False
+    else:
+        return not completed_at or is_datetime(completed_at)
+
+def is_put_task_data_valid(input):
+    data_types = {"title":str, "description":str}
+    for name, val_type in data_types.items():
+        if type(input.get(name)) != val_type:
+            return False
+    return True
+
+def is_datetime(string):
+    try:
+        x = datetime.strptime(string, '%a, %d %b %Y %H:%M:%S GMT')
+    except ValueError:
+        return False
+    return True
+
+# GOALS
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"], strict_slashes=False)
+def post_goals():
+    request_body = request.get_json()
+    if is_goal_data_valid(request_body):
+        new_goal = Goal(title=request_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+        return {"goal": new_goal.to_dict()}, 201
+    else:
+        return {"details": "Invalid data"}, 400
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_goals():
+    goals = Goal.query.all()
+    response = [goal.to_dict() for goal in goals]
+    return jsonify(response), 200
+
+@goals_bp.route("/<id>", methods=["GET"], strict_slashes=False)
+def get_goal(id):
+    goal = Goal.query.get(id)
+    if not goal:
+        return "", 404
+    return {"goal": goal.to_dict()}
+
+@goals_bp.route("/<id>", methods=["PUT"], strict_slashes=False)
+def put_goal(id):
+    goal = Goal.query.get(id)
+    if not goal:
+        return "", 404
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+    db.session.commit()
+    return {"goal": goal.to_dict()}, 200
+
+@goals_bp.route("/<id>", methods=["DELETE"], strict_slashes=False)
+def delete_goal(id):
+    goal = Goal.query.get(id)
+    if not goal:
+        return "", 404
+    goal = Goal.query.get(id)
+    db.session.delete(goal)
+    db.session.commit()
+    return {"details": f"Goal {str(goal.goal_id)} \"{str(goal.title)}\" successfully deleted"}, 200
+
+def is_goal_data_valid(input):
+    data_types = {"title":str}
+    for name, val_type in data_types.items():
+        if type(input.get(name)) != val_type:
+            return False
+    return True
+
+# GOALS x TASKS
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_goals_and_tasks(goal_id):
+    goal_id = int(goal_id)
+    if not Goal.query.get(goal_id):
+        return "", 404
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+    for task_id in task_ids:
+        task = Task.query.get(task_id)
+        task.goal_id = goal_id
+        db.session.commit()
+    all_task_ids = [task.task_id for task in Task.query.filter(Task.goal_id == goal_id)]
```

Comment:
Interesting!  This query brings up some neat questions (what if there were other tasks in the db with a matching goal_id, or what if some of the task ids that were passed in aren't valid).